Terraria/Net/Sockets/TcpSocket.cs

```diff
@@ -126,59 +126,27 @@ void ISocket.AsyncSend(byte[] data, int offset, int size, SocketSendCallback cal
 			this._connection.GetStream().BeginWrite(data, 0, size, new AsyncCallback(this.SendCallback), new Tuple<SocketSendCallback, object>(callback, state));
 		}
 
-		void Terraria.Net.Sockets.ISocket.Close()
+		void ISocket.Close()
 		{
-			lock (_connection)
-			{
-				this._connectionDisposed = true;
-				this._remoteAddress = null;
-				this._connection.Close();
-			}
+			this._remoteAddress = null;
+			this._connection.Close();
 		}
 
 		void Terraria.Net.Sockets.ISocket.Connect(RemoteAddress address)
 		{
-
 			TcpAddress tcpAddress = (TcpAddress)address;
 			this._connection.Connect(tcpAddress.Address, tcpAddress.Port);
 			this._remoteAddress = address;
-
-			lock (_connection)
```

Comment:
Was there a reason for these locks?
Were they placed by us?
It is not present in current Terraria code.